test/tools/jsontests/StateTests.cpp

```diff
@@ -41,6 +41,9 @@ namespace dev {  namespace test {
 
 void doStateTests(json_spirit::mValue& _v, bool _fillin)
 {
+	BOOST_REQUIRE_MESSAGE(!_fillin || _v.get_obj().size() == 1,
+		TestOutputHelper::testFileName() + " A GeneralStateTest filler should contain only one test.");
```

Comment:
Exactly.
* when `_fillin == true`, this assertion checks the number of tests in the file.
* when `_fillin == false`, this assertion is skipped.

The strict checking is required when doing the filling, not when running existing tests.